lib/types.ts

```diff
@@ -1,42 +1,30 @@
-// Common type aliases for better code readability
-
-import type {
-  Config as JestConfig,
-  TestResult as JestTestResult,
-} from '@jest/types';
-import type * as JestRunner from 'jest-runner';
-
-export type Milliseconds = JestTestResult.Milliseconds;
-export type Path = JestConfig.Path;
-
-export interface TestDetail {
-  title: string;
-  path: Path;
-}
+import type { TestContext, TestResult } from '@jest/test-result';
+import type { Config, TestRunnerOptions } from 'jest-runner';
 
 export interface CreateRunnerOptions<
-  ExtraOptionsType extends Record<string, unknown>,
+  ExtraOptions extends Record<string, unknown>,
 > {
-  getExtraOptions?: () => ExtraOptionsType;
+  getExtraOptions?: () => ExtraOptions;
 }
 
-// Copied and adapted from https://github.com/facebook/jest/blob/2dafb09d51584d3785f3280f569784ec4334b5d8/packages/jest-runner/src/index.ts#L48-L285
-export declare abstract class TestRunner {
-  readonly _globalConfig: JestConfig.GlobalConfig;
-
-  readonly _context: JestRunner.TestRunnerContext;
-
-  constructor(
-    globalConfig: JestConfig.GlobalConfig,
-    context?: JestRunner.TestRunnerContext,
-  );
+export type RunTestOptions<
+  ExtraOptions extends Record<string, unknown> = Record<string, unknown>,
+> = {
+  config: Config.ProjectConfig;
+  extraOptions: ExtraOptions;
+  globalConfig: Config.GlobalConfig;
+  rawModuleMap: ReturnType<TestContext['moduleMap']['getRawModuleMap']> | null;
+  options: TestRunnerOptions;
+  testPath: string;
+};
+
+export type RunTest<
+  ExtraOptions extends Record<string, unknown> = Record<string, unknown>,
+> = (
+  options: RunTestOptions<ExtraOptions>,
+) => TestResult | Error | Promise<TestResult | Error>;
```

Comment:
Wait.. Not sure about `Error` here. Is it really needed? One can throw error any way.

```suggestion
) => TestResult | Promise<TestResult>;
```